Libraries/libutil/Sources/MemoryFilesystem.cpp

```diff
@@ -158,6 +159,66 @@ isExecutable(std::string const &path) const
     return this->exists(path);
 }
 
+static Permissions
+AllPermissions()
+{
+    Permissions permissions;
+    permissions.user(Permissions::Permission::Read, true);
+    permissions.user(Permissions::Permission::Write, true);
+    permissions.user(Permissions::Permission::Execute, true);
+    permissions.group(Permissions::Permission::Read, true);
+    permissions.group(Permissions::Permission::Write, true);
+    permissions.group(Permissions::Permission::Execute, true);
+    permissions.other(Permissions::Permission::Read, true);
+    permissions.other(Permissions::Permission::Write, true);
+    permissions.other(Permissions::Permission::Execute, true);
+    return permissions;
+}
+
+ext::optional<Permissions> MemoryFilesystem::
+readFilePermissions(std::string const &path) const
+{
+    if (!this->isFile(path)) {
+        return ext::nullopt;
+    }
+
+    return AllPermissions();
+}
+
+ext::optional<Permissions> MemoryFilesystem::
+readSymbolicLinkPermissions(std::string const &path) const
+{
+    return ext::nullopt;
```

Comment:
Doesn't exist vs exists with all permissions. The memory filesystem doesn't support symlinks (yet)